mm2src/coins/solana/solana_common.rs

```diff
@@ -0,0 +1,59 @@
+use crate::solana::{SolanaAsyncCommonOps, SolanaCommonOps};
+use crate::{MarketCoinOps, WithdrawError, WithdrawRequest};
+use bigdecimal::ToPrimitive;
+use common::mm_error::MmError;
+use common::mm_number::BigDecimal;
+use futures::compat::Future01CompatExt;
+use solana_sdk::hash::Hash;
+use solana_sdk::native_token::LAMPORTS_PER_SOL;
+
+pub fn lamports_to_sol(lamports: u64) -> BigDecimal { BigDecimal::from(lamports) / BigDecimal::from(LAMPORTS_PER_SOL) }
+
+pub fn sol_to_lamports(sol: &BigDecimal) -> Option<u64> { (sol * BigDecimal::from(LAMPORTS_PER_SOL)).to_u64() }
+
+pub fn ui_amount_to_amount(ui_amount: BigDecimal, decimals: u8) -> Option<u64> {
+    (ui_amount * BigDecimal::from(10_u64.pow(decimals as u32))).to_u64()
+}
+
+pub fn amount_to_ui_amount(amount: u64, decimals: u8) -> BigDecimal {
+    BigDecimal::from(amount) / BigDecimal::from(10_u64.pow(decimals as u32))
+}
+
+pub async fn check_sufficient_balance<T>(
+    coin: &T,
+    req: &WithdrawRequest,
```

Comment:
> If you need to reuse it outside of the `withdraw` context, then you will have to construct the entire WithdrawRequest for no reason. It seems that this fn requires only the required numeric amount here. `WithdrawError` also seems excessive because it can return only a single variant of it.

Done - now return a specific Error, and the required parameters.